File: python/6764.py
def main():
    constanting = True
    constant = None

    increasing = True
    increase = None

    decreasing = True
    decrease = None

    for _ in range(4):
        number = int(input())

        if constant is None:
            constant = number
        elif number != constant:
            constanting = False

        if increase is None:
            increase = number
        elif number <= increase:
            increasing = False

        if decrease is None:
            decrease = number
        elif number >= decrease:
            decreasing = False

    if constanting:
        print('Constant Depth')
    elif increasing:
        print('Fish Rising')
    elif decreasing:
        print('Fish Diving')


if __name__ == '__main__':
    main()

File: rust/src/4/6764.rs
use std::io::stdin;

fn main() {
    let mut numbers = Vec::new();
    for _ in 0..4 {
        let mut input = String::new();
        stdin().read_line(&mut input).unwrap();
        let number = input
            .trim()
            .parse::<u32>().unwrap();
        numbers.push(number);
    }

    let mut increasing = true;
    let mut decreasing = true;
    let mut constant = true;

    for i in 1..4 {
        if numbers[i-1] >= numbers[i] {
            increasing = false;
        }
        if numbers[i-1] <= numbers[i] {
            decreasing = false;
        }
        if numbers[i-1] != numbers[i] {
            constant = false;
        }
    }

    if increasing {
        println!("Fish Rising");
    } else if decreasing {
        println!("Fish Diving");
    } else if constant {
        println!("Fish At Constant Depth");
    } else {
        println!("No Fish");
    }
}
